test/clj/puppetlabs/trapperkeeper/services/webserver/jetty9_service_test.clj

```diff
@@ -729,7 +712,34 @@
           (let [response (http-client-common/get async-client "http://localhost:8080/hello" {:as :text})]
             @in-request-handler
             (tk-app/stop app)
-            (is (not (nil? (:error @response))))))))))
+            (is (not (nil? (:error @response)))))))))
+
+  (testing "tk app can still restart even if stop timeout expires"
+    (let [in-request-handler (promise)
+          sleepy-service (tk-core/service
+                          [[:WebserverService add-ring-handler]]
+                          (init [this context]
+                                (add-ring-handler
+                                 (fn [_]
+                                   (deliver in-request-handler true)
+                                   (Thread/sleep 2000)
```

Comment:
Yeah that's true, it's fine for the first request to block, and it seems like it's also fine to wait to deliver the promise until after the restart has completed.